bats/tests/profile/sample.bats

```diff
@@ -32,69 +45,188 @@ start_app() {
     wait_for_container_engine
 }
 
-verify_settings() {
+verify_profiles() {
     PROFILE_TYPE=$PROFILE_LOCKED
     run profile_exists
-    "${assert}_success"
+    "${assert}_success" || return
 
     PROFILE_TYPE=$PROFILE_DEFAULTS
     run profile_exists
-    "${assert}_success"
+    "${assert}_success" || return
+}
 
+verify_settings() {
+    #settings from defaults profile
     run get_setting .containerEngine.name
-    "${assert}_output" "$PROFILE_CONTAINER_ENGINE"
+    "${assert}_output" "$PROFILE_CONTAINER_ENGINE" || return
 
     run get_setting .application.startInBackground
-    "${assert}_output" "$PROFILE_START_IN_BACKGROUND"
+    "${assert}_output" "$PROFILE_START_IN_BACKGROUND" || return
+    #settings from locked profile
+    run get_setting .containerEngine.allowedImages.enabled
+    "${assert}_output" "$PROFILE_USE_IMAGE_ALLOW_LIST" || return
+
+    run get_setting .containerEngine.allowedImages.patterns
+    "${assert}_output" --partial "${PROFILE_IMAGE_PATTERNS[@]}" || return
+
+    run get_setting .application.extensions.allowed.enabled
+    "${assert}_output" "$PROFILE_USE_EXTENSION_LIST" || return
+
+    run get_setting .application.extensions.allowed.list
+    "${assert}_output" --partial "${PROFILE_EXTENSION_LIST[@]}" || return
+
+    run get_setting .kubernetes.version
+    "${assert}_output" "$PROFILE_LOCKED_KUBERNETES_VERSION" || return
+    refute_output "$PROFILE_DEFAULTS_KUBERNETES_VERSION" || return
 }
 
 @test 'initial factory reset' {
     factory_reset
 }
 
-@test 'start up without profile' {
+@test 'start up with NO profiles' {
     RD_USE_PROFILE=false
-    start_app
+    RD_USE_IMAGE_ALLOW_LIST=false
```

Comment:
Because of line 117 from `vm.bash` `add_profile_bool containerEngine.allowedImages.enabled "$image_allow_list"` The images list enabled/disabled is stored in the profile every time `start_container_engine` is called. So, as in this test we are dealing with profiles most of the time, I set this env variable to true, but for the particular test that starts Rancher Desktop with no profiles, I want it to be indeed disabled, so I set it to false just for this particular test